src/challenges/challengeOne/4.Form/Form.test.tsx

```diff
@@ -1,8 +1,81 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Form } from "./Form";
-
-// describe("Form", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { Form, FormProps, FormField } from "./Form";
+
+describe("Form", () => {
+  const fields: FormField[] = [
+    {
+      label: "Name",
+      type: "text",
+      name: "name",
+      placeholder: "Enter your name",
+      value: "",
+      id: "name",
+    },
+    {
+      label: "Email",
+      type: "email",
+      name: "email",
+      placeholder: "Enter your email",
+      value: "",
+      id: "email",
+    },
+  ];
+
+  const handleSubmit = jest.fn();
```

Comment:
A handler function is slightly different from a callback function in React. So we name them differently. a

A callback is named with `on`, so `onClick`, `onToggle`, `onClose`. These are functions passed into other functions or components in React's case. 

Handle functions orchestrate functionality inside a component, and often use callbacks inside their execution.

```
const handleSubmit  = () => {
  onSubmit();
  setCondition(true);
}
```

That aside, you should add `mock` to the beginning of the name of the variable so the developer can know immediately it's a mock of the `onSubmit` function.

`mockOnSubmit`